Sources/Private/Experimental/Extensions/KeyframeGroup+CAKeyframeAnimation.swift

```diff
@@ -60,8 +64,27 @@ extension KeyframeGroup {
       keyTimes.append(NSNumber(value: Float(keyframe.keyTime)))
     }
 
+    // Validate that we have correct start (0.0) and end (1.0) keyframes.
+    //
+    // From the documentation of `CAKeyframeAnimation.keyTimes`:
+    //
+    //   If the `calculationMode` is set to `linear` or `cubic`, the first value in the array
+    //   must be 0.0 and the last value must be 1.0. All intermediate values represent
+    //   time points between the start and end times.
+    //
+    if keyTimes.first != 0.0 {
+      keyTimes.insert(0.0, at: 0)
+      values.insert(values.first!, at: 0)
+    }
+
+    if keyTimes.last != 1.0 {
+      keyTimes.append(1.0)
+      values.append(values.last!)
```

Comment:
This was quite the tricky issue I ran in to -- the documentation says "the first value in the array must be 0.0 and the last value must be 1.0". 

This doesn't seems _strictly required_ at runtime (because it worked correctly at runtime in the previous PR) **BUT** the `layer.presentation()` copy seems to silently ignore any keyframes that don't follow this rule.

Reply:
Interesting! Core Animation is full of so many weird things like this 😅

Reply:
![image](https://user-images.githubusercontent.com/1811727/146445716-b412b0b9-26e3-43bd-8a87-4f4e6e9b12f5.png)